src/app/[locale]/uikit/WeekView.tsx

```diff
@@ -0,0 +1,197 @@
+"use client";
+
+import {
+  addDays,
+  addMonths,
+  differenceInCalendarMonths,
+  differenceInDays,
+  eachMonthOfInterval,
+  isPast,
+  isToday,
+  lightFormat,
+  startOfMonth,
+  startOfWeek,
+} from "date-fns";
+import { memo, useCallback, useState } from "react";
+import { GrNext } from "react-icons/gr";
+import { Button, Select } from "@/components/uikit";
+import { Locale } from "@/navigation";
+import { css } from "~/styled-system/css";
+import { Box, Grid, GridItem } from "~/styled-system/jsx";
+
+export function WeekView({
+  locale,
+  todayDate,
+}: {
+  locale: Locale;
+  todayDate: Date;
+}) {
+  const [originDate, setOriginDate] = useState(todayDate);
+
+  const changeWeek = (direction: "prev" | "next") => {
+    return () =>
+      setOriginDate((prevOrigin) => {
+        const multiplier = direction === "prev" ? -1 : 1;
+
+        return addDays(prevOrigin, 7 * multiplier);
+      });
+  };
+
+  const changeMonth = useCallback((monthCount: number) => {
+    const nextOriginDate = startOfMonth(addMonths(todayDate, monthCount));
+
+    setOriginDate(isPast(nextOriginDate) ? todayDate : nextOriginDate);
+  }, []);
+
+  const weekViewData = getWeekViewData({ locale, originDate });
+
+  return (
+    <Box>
+      <MonthSelect
+        locale={locale}
+        month={differenceInCalendarMonths(originDate, todayDate)}
+        changeMonth={changeMonth}
+        todayDate={todayDate}
+      />
+      <Grid
+        gridTemplateColumns="auto 1fr auto"
+        // TODO adjust layout to use common Grid parent for all the controls
+        gridTemplateAreas="'prev-week . next-week'"
+      >
+        <GridItem gridArea="prev-week">
+          <Button
+            type="button"
+            variant="ghost"
+            onClick={changeWeek("prev")}
+            disabled={isPast(originDate)}
+          >
+            <GrNext size={30} className={css({ rotate: "180deg" })} />
+          </Button>
+        </GridItem>
+
+        <GridItem gridArea="next-week">
+          <Button
+            type="button"
+            variant="ghost"
+            onClick={changeWeek("next")}
+            disabled={
+              differenceInCalendarMonths(addDays(originDate, 7), todayDate) > 12
+            }
```

Comment:
этот проп лучше доставать из хука тоже, чтобы компоненту не усложнять 
```suggestion
            disabled={canNextWeek}
```